ios/PresentationLayer/UIToolkit/Sources/UIToolkit/BaseViews/SwiftUI/Image/RemoteImage.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  Created by David Kadlček on 12.05.2023
+//  Copyright © 2023 Matee. All rights reserved.
+//
+
+import Foundation
+import SwiftUI
+
+public struct RemoteImage: View {
+    
+    @State private var image: Image?
+    private let stringURL: String?
+    private let defaultImage: Image
+    private let contentMode: ContentMode
+    private let failureMessage: String?
+    
+    public init(
+        stringURL: String?,
+        defaultImage: Image,
+        contentMode: ContentMode = .fit,
+        failureMessage: String? = nil
+    ) {
+        self.stringURL = stringURL
+        self.defaultImage = defaultImage
+        self.contentMode = contentMode
+        self.failureMessage = failureMessage
+    }
+    
+    public var body: some View {
+        if let image {
+            image
+                .resizable()
+                .aspectRatio(contentMode: contentMode)
+        } else if let stringURL = stringURL, let url = URL(string: stringURL) {
+            placeholderContent()
+                .skeleton(true)
+                .onAppear(perform: { downloadRemoteImage(from: url) })
+        } else {
+            placeholderContent()
```

Comment:
Isn't `failureContent` supposed to be here? 🤔

Reply:
Yep, you are right.